### PR TITLE
Upgrade lib to 2.20

### PR DESCRIPTION
**Pull request by huihuiy02**
customerio/issues#3854

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -27,6 +28,18 @@ commands:
...
             make test
+            deactivate
+      - run:
+          name: reinstall the requests to the latest version
+          command: |
+            python -m venv venv
+            . venv/bin/activate
+            pip install --upgrade requests
+      - run:
+          name: run tests again for the minor version
```
Comment (jatinn) [inline .circleci/config.yml]: latest not minor version now. repeats for the others below as well

File: Makefile [hunk 2]
```diff
@@ -20,5 +20,6 @@ upload:
 	echo "*** Now upload the binary to PyPi *** (one second)" && sleep 3 && open dist & open "http://pypi.python.org/pypi?%3Aaction=pkg_edit&name=customerio" # python setup.py upload
 
 test:
-	openssl req -new -newkey rsa:1024 -days 10 -nodes -x509 -subj "/C=CA/ST=Ontario/L=Toronto/O=Test/CN=127.0.0.1" -keyout ./tests/server.pem -out ./tests/server.pem
+	openssl version -a
```
Comment (jatinn) [inline Makefile]: any particular reason to print the version info?

File: tests/server.py [hunk 3]
```diff
@@ -77,10 +76,12 @@ def setUpClass(cls):
         cls.server = HTTPServer(("localhost", 0), Handler)
         # hack needed to setup ssl server
         ssl.wrap_socket = sslwrap(ssl.wrap_socket)
+
         # upgrade to https
         cls.server.socket = ssl.wrap_socket(cls.server.socket,
             certfile='./tests/server.pem',
             server_side=True)
+            # ssl_version=ssl.PROTOCOL_TLSv1)
```
Comment (jatinn) [inline tests/server.py]: leftover from testing?

**Comment by huihuiy02** [comment]
I have cleaned up all the debug stuff.

**Pull request merged**